src/tables.ts

```diff
@@ -586,6 +718,41 @@ export class Tables {
     return clefs[clef];
   }
 
+  /**
+   * Use the provided key to look up a value in this font's metrics file (e.g., bravuraMetrics.ts, petalumaMetrics.ts).
+   * @param key is a string separated by periods (e.g., stave.endPaddingMax, clef.lineCount.'5'.shiftY).
+   * @param defaultValue is returned if the lookup fails.
+   * @returns the retrieved value (or `defaultValue` if the lookup fails).
+   */
+  // eslint-disable-next-line
+  static lookupMetric(key: string, defaultValue?: Record<string, any> | number): any {
```

Comment:
Was this previously in the Font class as `Font.lookupMetric()`?

I'm a bit worried about tables.ts becoming a huge catch-all class that handles everything.

Should the `CommonMetrics` class be in its own file? It could be called the `Metrics` class, and placed in a `metrics.ts` file. All the CommonMetrics serve as the defaults that can be overridden by other fonts.

Then this lookupMetric(...) function can be inside metrics.ts as Metrics.lookup().

Anyways, just a thought. If you believe that the Tables is in fact the Metrics class, then it's OK to keep it all here. I just want to make sure one class isn't handling everything.

Reply:
OK, thanks. My main concern is having tables.ts grow to handle all the things.... :-)

And maybe tables.ts needs a better name? Or is "Tables" the best description for its purpose that we can come up with, hehe....